### PR TITLE
Fix iframe display for Firefox

### PR DESCRIPTION
**Pull request by drwpow**
## Reason for change

Fixes https://github.com/manifoldco/engineering/issues/9089.

### Solution

@seriousben and I observed in Firefox when an `<iframe>` element is `display: none`, Render’s React app wasn’t rendering (which Render needed to complete the OAuth dance). I was able to confirm that _some_ JavaScript was executing, but not all of it. Not sure if it’s a performance thing or a security thing or a mix of both. But whatever the reason, the end user result was **our web components didn’t work in Firefox.** So a pretty big deal.

This adds `display: block` which is essential, but along with that it adds other inline properties (for specificity) that ensure that the iframe takes up space, but isn’t visible to the user and doesn’t affect the rest of the page at all. 

This has been tested locally and in my testing seemed to completely solve the problem It hasn’t failed to authenticate once.

This problem also seemed to be unique to Firefox—Chrome and Safari seemed to work—but again, Firefox was almost completely broken.

## Testing

Difficult to test, but here’s what I did locally and verified it worked:

- Shadowcat: run `npm run build && cp -r dist /local/path/to/manifoldco/ui/node_modules/@manifoldco/shadowcat`
- UI run `npm run build && cp -r dist /local/path/to/manifoldco/dashboard-render/node_modules/@manifoldco/ui`
- Render Dashboard: restart the server. The newly-built Shadowcat should have shipped within the newly-built UI (I know `yarn link` is a thing, but it kept pulling old, cached versions of shadowcat).

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    'import/no-extraneous-dependencies': 'off', // We need zero deps for npm
+    'import/prefer-default-export': 'off', // named exports are perfectly fine
+    'lines-between-class-members': 'off', // class members don’t need that space!
+    'max-len': 'off', // let Prettier decide
+    'object-curly-newline': 'off', // let Prettier decide,
+    'prettier/prettier': 'error',
+    'no-console': ['error', { allow: ['warn', 'error'] }],
+    'func-names': 'off',
+  },
+};
```
Comment (drwpow) [inline .eslintrc.js]: Why wasn’t ESLint set up on this project?

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "printWidth": 100,
+  "singleQuote": true,
+  "trailingComma": "es5"
+}
```
Comment (drwpow) [inline .prettierrc:5]: Prettier was missing as well.

File: README.md [hunk 3]
```diff
@@ -19,30 +19,6 @@ defineCustomElements(window);
...
 ```
 
-4. Listen for the event `tokenReceived`:
-
-```js
-document.addEventListener("tokenReceived", e => {
-  console.log(e.detail); // received token
-  // do stuff...
-});
-```
```
Comment (drwpow) [inline README.md]: These instructions weren’t accurate; currently it only emits events perceptible to other Stencil libraries.

File: README.md [hunk 4]
```diff
@@ -19,30 +19,6 @@ defineCustomElements(window);
...
-1. Open the console. There will be a message/token logged.
-
-## Publish to npm
-
-1. Add a git tag with the version number you want to publish. 
-   - For example, version 1 would be `git tag v1.0.0`.
-1. Publish the version to npm by running `npm run publish`. 
-   - Ensure that you're logged into your npm account and have access to publish packages in the `@manifoldco` org when you do this.
-
-
```
Comment (drwpow) [inline README.md]: Let’s not ship internal dev notes to npm, especially when we refer to this README for instructions.

File: README.md [hunk 5]
```diff
@@ -1,16 +1,16 @@
 # shadowcat
 
-Invisible web component auth implementation of Manifold&#39;s PUMA &lt;https://github.com/manifoldco/puma&gt;
```
Comment (drwpow) [inline README.md]: We were linking to PUMA, a private repo, from this public one.

File: package.json [hunk 6]
```diff
@@ -1,6 +1,12 @@
 {
   "name": "@manifoldco/shadowcat",
   "description": "Invisible web component auth implementation of Manifold's PUMA",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/manifoldco/ui.git"
+  },
+  "author": "manifoldco",
+  "license": "BSD",
```
Comment (drwpow) [inline package.json]: License was incorrect, and missing `repository` field was raising install warnings in UI

File: README.md (hunk 3 above)
Comment (sslotsky) [reply README.md]: Well, that's not true, is it? You're right that the event doesn't bubble to the document, but you can listen for the event on the element. It should probably just be changed to show a query for the `manifold-oauth` component and do `component.addEventListener` instead.

**Review by sslotsky — CHANGES_REQUESTED** [review]
Can you please make a separate PR for all the things that are unrelated to the problem we're solving with this one?

File: README.md (hunk 3 above)
Comment (drwpow) [reply README.md]: Well the event name is wrong, too. It’s `receiveManifoldToken`, but I had to dig into the Netlify preview app which wasn’t named correctly.

Reason I’m being a stickler is it’s not for me; it’s for our partners. If the docs are wrong, then as far as they’re concerned it’s broken software. This is broken.

Comment (sslotsky) [reply README.md]: Sure but does that mean the documentation should be removed, or fixed?

**Comment by drwpow** [comment]
Moved to separate PRs.

File: config.yml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+version: 1
```
Comment (sslotsky) [inline config.yml:1]: What's this config?

File: src/components/manifold-oauth/manifold-oauth.tsx [hunk 8]
```diff
@@ -1,35 +1,57 @@
...
-  tag: "manifold-oauth"
-})
+@Component({ tag: 'manifold-oauth' })
 export class ManifoldOauth {
-  @Prop() oauthUrl?: string = "https://login.manifold.co/signin/oauth/web";
+  @Prop() oauthUrl?: string = 'https://login.manifold.co/signin/oauth/web';
   @Event() receiveManifoldToken: EventEmitter<AuthToken>;
 
-  private loadTime?: Date;
+  private loadTime?;
```
Comment (sslotsky) [inline src/components/manifold-oauth/manifold-oauth.tsx]: Why was the type removed?

File: config.yml (hunk 7 above)
Comment (drwpow) [reply config.yml:1]: dependabot ignore. When you run tests, Stencil re-installs old Jest. Dependabot tries to re-install newer Jest. This reduces diff noise.

File: src/components/manifold-oauth/manifold-oauth.tsx (hunk 8 above)
Comment (drwpow) [reply src/components/manifold-oauth/manifold-oauth.tsx]: Oh that was an ESLint rule suggestion from @typescript-eslint. I can revert it for now and fix it in the ESLint PR

**Review by sslotsky — APPROVED** [review]

**Pull request merged**